servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/publisher/TimeoutPublisherTest.java

```diff
@@ -286,11 +307,14 @@ public Cancellable execute(final Runnable task) throws RejectedExecutionExceptio
     }
 
     private void init() {
-        init(publisher, true);
+        init(publisher, true, untilTermination);
     }
 
-    private void init(Publisher<Integer> publisher, boolean expectOnSubscribe) {
-        toSource(publisher.idleTimeout(1, NANOSECONDS, testExecutor)).subscribe(subscriber);
+    private void init(Publisher<Integer> publisher, boolean expectOnSubscribe, boolean untilTermination) {
+        publisher = untilTermination ?
+                publisher.withTimeout(ONE_NANOSECOND, testExecutor)
```

Comment:
Consider adding a test that verifies the `onNext` does not update the timeout

Reply:
This task has proved to be more difficult than I expected. I was able to verify that it works as expected with a simple program using realistic timeouts but I cannot get it to behave properly using the TestExecutor et al fixtures.